CeeFM/Decideify
# Movies.js Algorithm and Cleanup Work Part 1

**Description**
This pull request is the first in a series that work on finalizing V1 of our suggestion algorithm, browser rendering, and UI/UX functionality for the Movies hub / movies.js component. We have our genre dropdown menu working now, and submitting it will both manipulate state and data on the page to prepare for a random movie suggestion.

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- On the movies.js page:
  - You can choose a genre / type for the piece of content from a dropdown menu on the movie hub page, and then test that submission and see two things happen
    - 1) the "CategoryId" property of the "suggestion" state variable should be updated to the Id of the option from the "Category" database table for the option you chose
    - 2) the entire Category object that you selected should also be printed to the console after being saved to another state variable named userCategory  

Checklist
